website/blog/2023-08-14-mongodb-sorting-scalar.md

```diff
@@ -0,0 +1,166 @@
+---
+slug: mongodb-sorting-scalar
+title: How MongoDB Sorting Works for Scalar Values
+authors: [chi]
+description: >
+  In this blog post, we explore how MongoDB sorting works for scalar values.
+keywords:
+  - MongoDB sorting
+  - BSON comparison order
+image: /img/blog/post-cover-image.jpg
+tags: [community, product, tutorial]
+---
+
+![MongoDB sorting works for scalar values](/img/blog/mongodb-sorting-scalar.jpg)
+
+In this blog post, we explore how MongoDB sorting works for scalar values.
+
+<!--truncate-->
+
+Sorting compares BSON values to determine which value is equal, greater or less than the other to order them in the ascending or descending order.
+For comparing different BSON types, [BSON comparison order](#bson-comparison-order) is used.
+
+## BSON comparison order
+
+If two BSON values share the same BSON type, their values are compared to determine which value is greater or less.
+However, if the BSON types are different, a predefined BSON comparison order is used to determine which BSON type is greater or less.
```

Comment:
I'm not sure it's fully clear when we say "which BSON type is greater or less"... We still want to understand which value is greater or less, but as they are of different types, we use comparison order for types first. Probably the current statement is also straightforward having in mind the rest of the context, so feel free to ignore my comment.